### ISSUE TITLE
Update development instructions

### ISSUE THREAD

**Issue by WithoutPants** [stashapp/stash#78]
I've been attempting to build stash on Windows and have had a great deal of trouble getting it running due to a number of issues.

The build process requires packr2 to be installed. I ran `go get -u github.com/gobuffalo/packr/v2/packr2` which did the job, but I think it also added a bunch of dependencies to the go.mod file. The prerequisite (and correct procedure) needs to be documented in the Development section of the readme.

Before I worked out how to install packr2, I was running `go build` instead, and this basically worked. The issue I discovered was that the UI had been updated (to add the link anchors to the File Info tab), and no matter what I did, I couldn't get the UI to reflect these changes. I discovered that this is because the packed-packr.go file is in the repository and was not updated with the recent UI changes. The packr2 documentation does note the issues with checking in the generated files:

> NOTE: It is not recommended to check-in these generated -packr.go files. They can be large, and can easily become out of date if not careful. It is recommended that you always run packr2 clean after running the packr2 tool.

If the packed-packr.go file had not been in the repository, the direct build would have failed because packr2 hadn't been run, which to my mind is the correct behaviour.

For Windows, MingW is required for its use of gcc, but make doesn't really work for building. On my particular machine (which is 32 bit) MingW provides mingw32-make.exe. I don't know if the 64 bit version is named make.exe or not, but it was something that also tripped me up. Finally, the Makefile itself does not work with Windows. The Makefile assumes a bash (or similar) shell, not cmd, so the `CGO_ENABLED=1` part causes an error. This wasn't a big issue, since I just ran the command manually, but it should probably be documented (or fixed if there is a way to work around it).

So, with packr2 installed, I ran the command in the Makefile under the build target. When running the exe and navigating to the correct URL, I got an empty response. It turned out that packr hadn't packed any of the UI files. I only discovered this by doing a diff of the packed-packr.go file, and then looking through to find how packr discovers and builds this file. I finally discovered that the UI needed to be built. This isn't documented in the readme file. It does say that Yarn is required, but doesn't say that `yarn run build` is required to be run in the ui/v2 directory. The UI build process needs to documented at least, and/or added as a target to the Makefile. It may also be useful to document that the UI needs to be (re)built if the server returns an empty file, or if the UI is outdated.

To summarise, I propose the following:

- document the process of installing packr2, since this is required to build
- remove packed-packr.go from the repository and add it to .gitignore
- document the Windows procedure to build and/or fix the Makefile to be compatible with Windows command prompt
- add the UI build process to a new target in the Makefile and/or add the UI build procedure to the readme.

**Comment by StashAppDev** [comment]
I've addressed some of these in the latest commits.  I'll need to spend time on my windows machine in order to update the Makefile and ensure it works properly
